django/db/backends/base/base.py

```diff
@@ -170,7 +170,7 @@ def init_connection_state(self):
         """Initialize the database connection settings."""
         raise NotImplementedError('subclasses of BaseDatabaseWrapper may require an init_connection_state() method')
 
-    def create_cursor(self, name=None):
+    def create_cursor(self, name=None, *args, **kwargs):
```

Comment:
this is for using underlying database driver so I don't think that would be an issue. but what do you suggest?

Reply:
I would have to know more about the expected use cases to provide a true recommendation. But an example of an alternative is--

```python
def create_cursor(self, name=None, args=None, kwargs=None):
```

I won't list all of the advantages of this alternative. But to give you an idea, some of the advantages are that (1) it supports adding more keyword arguments to `create_cursor()` without breaking backwards compatibility, so it is "future-proof," (2) it permits `name` to be in `kwargs`, (3) it allows `name` to be provided as a keyword argument even if `*args` are provided, and (4) providing `*args` doesn't require `name` to be passed.

Reply:
thanks for your input, have a look on the description of the ticket https://code.djangoproject.com/ticket/3461 and share your thoughts

Reply:
Okay, I skimmed it. I would recommend what I suggested as a starting point. It would work in full generality (no matter what the API is at the lower level), while having the advantages I listed. I can continue to think, though. It's possible another alternative could involve using `functools.partial()` [a la `asyncio`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-pass-keywords), but I would need to think a bit more.

Reply:
the pr is too old, and the ticket is 10 years, so I really want to finish this off for 2.1 soon :dagger:

Reply:
OK will update the code as suggested